components/data_stores/actions/add-update-multiple-records/add-update-multiple-records.mjs

```diff
@@ -23,82 +24,81 @@ export default {
-      } else if (typeof obj === "object") {
-        result = {
-          ...result,
-          ...obj,
-        };
-      } //else pass, because it is neither object nor array
-      return result;
+    /**
+     * Tries to convert a string to a JSON object or a primitive value.
+     * @param {any} value Value to be converted
+     * @returns The converted value
+     */
+    convertString(value) {
+      // If type is already primitive non string, return it
+      if ([
+        "boolean",
+        "number",
+      ].includes(typeof value)) {
+        return value;
+      }
+
+      // Tries to convert string to possible other primitive types
+      if (typeof value === "string") {
+        // Convert boolean
+        if (value.toLowerCase() === "true") {
+          return true;
+        }
+        if (value.toLowerCase() === "false") {
+          return false;
+        }
+
+        // Convert number
+        if (!isNaN(value)) {
+          return parseFloat(value);
+        }
+      }
+
+      // Try to evaluate string as javascript, using xss as extra security
+      // If some problem occurs, return the original string
+      try {
+        return eval(`(${xss(value)})`);
+      } catch {
+        return value;
+      }
     },
-    //Handling all serialized and unserialized inputs
-    //to treat values as objects no matter how they entered.
-    _parseData(obj) {
-      if (typeof obj === "string") {
-        const sanitizedValue = this.data_stores.sanitizeJson(obj);
-        const json = JSON.parse(sanitizedValue);
-        return this._parseDataRecursive(json);
-      } else if (typeof obj === "object") {
-        const newObj = {};
-        for (const [
-          key,
-          value,
-        ] of Object.entries(obj)) {
-          if (typeof value === "string") {
-            try {
-              const sanitizedValue = this.data_stores.sanitizeJson(value);
-              const json = JSON.parse(sanitizedValue);
-              if (typeof json !== "object") {
-                newObj[key] = json;
-              } else {
-                newObj[key] = this._parseDataRecursive(json);
-              }
-            } catch (err) {
-              newObj[key] = value;
-            }
-          } else if (typeof value === "object") {
-            newObj[key] = this._parseDataRecursive(value);
-          } else {
-            newObj[key] = value;
-          }
+    /**
+     * Add all the key-value pairs in the map object to be used in the data store
+     * @param {*} data Data to be parsed
+     * @param {Map} map Hashmap to be updated
+     */
+    populateHashMapOfData(data, map) {
+      if (!Array.isArray(data) && typeof(data) === "object") {
+        const keys = Object.keys(data);
+        for (const key of keys) {
+          map[key] = this.convertString(data[key]);
+        }
+        return;
+      }
+
+      if (Array.isArray(data)) {
+        for (const item of data) {
+          this.populateHashMapOfData(item, map);
         }
```

Comment:
Can this be the first check in the method?
Then checking if `!Array.isArray(data)` in the other condition is not necessary

```suggestion
      if (Array.isArray(data)) {
        data.forEach((item) => this.populateHashMapOfData(item, map));
```

Reply:
Not really, I need these two `ifs` due to some nested objects and because of the recursivity. Also, If it's not an array or object, it will be ignored for the final map